### PR TITLE
fix "cannot link" bug with import target containers

### PR DESCRIPTION
**Pull request by mtbc**
Fixes https://trello.com/c/kWgFImSC/102-bug-import-target by not considering among the candidate import targets any container that the user cannot link to.

### CODE REVIEW

File: src/main/java/ome/formats/importer/targets/ModelImportTarget.java [hunk 1]
```diff
@@ -131,7 +130,13 @@ public IObject load(OMEROMetadataStoreClient client, ImportContainer ic) throws
...
                 + " order by o.id " + order,
                 new ParametersI().add("name", rstring(name)));
-            if (objs.size() == 0 || discriminator.startsWith("@")) {
+            final Iterator<IObject> objIter = objs.iterator();
+            while (objIter.hasNext()) {
+                if (!objIter.next().getDetails().getPermissions().canLink()) {
+                    objIter.remove();
+                }
+            }
+            if (objs.isEmpty() || discriminator.startsWith("@")) {
```
Comment (joshmoore) [inline src/main/java/ome/formats/importer/targets/ModelImportTarget.java]: Perhaps skip the query completely if the discriminator is `@`?

Comment (mtbc) [reply src/main/java/ome/formats/importer/targets/ModelImportTarget.java]: I had thought about that. :smiley: I was going for a minimal change but can certainly be more invasive.

**Comment by pwalczysko** [comment]
Old == without this PR
New == with this PR

Old client connecting to new server -> Fail
New client connecting to old server -> Pass
New client connection to new server -> Pass

So the fix is rather in the client, not in the server.

The fix works, tried both manual -T as well as bulk import of idr0021.

Ready to merge FMPOV

**Comment by joshmoore** [comment]
Very exciting. Thank you all!

**Comment by joshmoore** [comment]
https://web-proxy.openmicroscopy.org/west-ci/job/OMERO-test-integration/349/testReport/junit/OmeroPy.test.integration.clitest.test_import/TestImport/testImportLinkableTarget_False_False_/ is nicely green.

**Pull request merged**